tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -1391,6 +1392,49 @@ def test_get_datasets_nonexistent_collection_404(self):
             self.assertEqual(404, response.status_code)
 
 
+class TestGetDatasetIdVersions(BaseAPIPortalTest):
+    def test_get_dataset_id_versions_ok(self):
+        collection = self.generate_published_collection()
```

Comment:
Add more documentation. It's obscure that your are testing that unpublished datasets should not appear in the response.